buildpacks/builder.go

```diff
@@ -125,26 +111,28 @@ func (builder *Builder) Build(ctx context.Context, f fn.Function) (err error) {
 		}
 	}
 
+	var isTrustedBuilderFunc = func(b string) bool {
+		return !daemonIsPodmanBeforeV330 &&
+			(strings.HasPrefix(packBuilder, "quay.io/boson") ||
+				strings.HasPrefix(packBuilder, "gcr.io/paketo-buildpacks"))
+	}
 	packOpts := pack.BuildOptions{
 		AppPath:        f.Root,
 		Image:          f.Image,
 		LifecycleImage: "quay.io/boson/lifecycle:0.12.0",
 		Builder:        packBuilder,
 		Env:            buildEnvs,
 		Buildpacks:     f.Buildpacks,
-		TrustBuilder: !daemonIsPodmanBeforeV330 &&
-			(strings.HasPrefix(packBuilder, "quay.io/boson") ||
-				strings.HasPrefix(packBuilder, "gcr.io/paketo-buildpacks")),
-		DockerHost: dockerHost,
+		TrustBuilder:   isTrustedBuilderFunc,
+		DockerHost:     dockerHost,
 		ContainerConfig: struct {
 			Network string
 			Volumes []string
 		}{Network: network, Volumes: nil},
 	}
 
-	dockerClientWrapper := &clientWrapper{cli}
 	// Client with a logger which is enabled if in Verbose mode.
-	packClient, err := pack.NewClient(pack.WithLogger(logging.New(logWriter)), pack.WithDockerClient(dockerClientWrapper))
+	packClient, err := pack.NewClient(pack.WithLogger(logging.NewSimpleLogger(logWriter)))
```

Comment:
@Shashankft9  custom docker client `cli` still should be set in order to support `SSH`.